azure-pipelines.yml

```diff
@@ -0,0 +1,164 @@
+
+              Write-Host " (i) Updated pubspec.yaml content: $newPubspecContent"
+
+              $newPubspecContent | Set-Content -Path "$(Build.SourcesDirectory)/device_calendar/pubspec.yaml"
+
+        - task: PowerShell@2
+          displayName: 'Pre-release versioning - CHANGELOG.md'
+          inputs:
+            targetType: 'inline'
+            script: |
+              [string] $changelogContent = Get-Content -Path "$(Build.SourcesDirectory)/device_calendar/CHANGELOG.md" -Raw
+
+              Write-Host " (i) Current version: $env.currentVersion"
+              Write-Host " (i) Original CHANGELOG.md content: $changelogContent"
+
+              [string] $newChangelogContent = $changelogContent -replace $env:currentVersion, "$env:currentVersion-dev.$(Build.BuildID)"
+
+              Write-Host " (i) Updated CHANGELOG.md content: $newChangelogContent"
+
+              $newChangelogContent | Set-Content -Path "$(Build.SourcesDirectory)/device_calendar/CHANGELOG.md"
+
+        - task: CmdLine@2
+          displayName: 'Dry run publish'
+          inputs:
+            workingDirectory: '$(Build.SourcesDirectory)/device_calendar'
+            script: '$(flutterExecPath) pub publish --dry-run'
+
+        - task: CmdLine@2
+          displayName: 'Publish'
+          inputs:
+            workingDirectory: '$(Build.SourcesDirectory)/device_calendar'
+            script: '$(flutterExecPath) pub publish --force'
+            
+  # ----------- CD Production -----------
+  - stage: ReleaseProd
+    dependsOn: Build
+    condition: and(succeeded(), eq(variables['Build.SourceBranch'], 'refs/master'))
+    jobs: 
+     - job: Release
+       pool:
+        vmImage: 'windows-latest'
+       steps:
+        - task: FlutterInstall@0
+          displayName: 'Flutter install'
+          inputs:
+            channel: 'stable'
+            version: 'latest'
+
+        - task: CmdLine@2
+          displayName: 'Dry run publish'
+          inputs:
+            workingDirectory: '$(Build.SourcesDirectory)/device_calendar'
+            script: '$(flutterExecPath) pub publish --dry-run'
+
+        - task: CmdLine@2
+          displayName: 'Publish'
+          inputs:
+            workingDirectory: '$(Build.SourcesDirectory)/device_calendar'
+            script: '$(flutterExecPath) pub publish --force'
+
+        - task: PowerShell@2
+          displayName: 'Get version from pubspec.yaml'
+          inputs:
+            targetType: 'inline'
+            script: |
+              [string] $version = select-string -path "$(Build.SourcesDirectory)/device_calendar/pubspec.yaml" -pattern "(?<=version: ).*" | %{ $_.Matches[0].Value }
+              
+              Write-Host "##vso[task.setvariable variable=currentVersion]$version"
+
+        - task: GitHubRelease@1
+          displayName: 'Tag a release to GitHub'
+          inputs:
+            gitHubConnection: 'github.com_Brett09'
```

Comment:
Shouldn't we have like a service account for this task, instead of using a personal one?

Reply:
Nick has mentioned it already and will go through that with me